pynestml/symbols/predefined_functions.py

```diff
@@ -168,9 +168,9 @@ def __register_exponent_function(cls):
         Registers the exponent (e(X)) function.
         """
         params = list()
-        params.append(PredefinedTypes.get_real_type())  # the argument
+        params.append(PredefinedTypes.get_template_type(0))  # the argument
         symbol = FunctionSymbol(name=cls.EXP, param_types=params,
-                                return_type=PredefinedTypes.get_real_type(),
+                                return_type=PredefinedTypes.get_template_type(0),
```

Comment:
in the case of the exponential function (and almost all non-polynomial function), it is correctly defined only if both the entry type and the return type are real.
I.e. in this case the previous code should be kept.